### PR TITLE
Support create(), update(), and delete() methods.

### PR DESCRIPTION
**Pull request by mckoss**
@tomlarkworthy @davideast

### CODE REVIEW

File: samples/chat.bolt [hunk 1]
```diff
@@ -12,7 +12,10 @@ path /posts/{roomid}/{postid} is Timestamped<Post> {
 
 type RoomInfo {
   read() { isMember(this) || isUser(this.creator) }
-  write() { isInitial(this) || isUser(this.creator) }
+
+  create() { true }
+  update() { isUser(prior(this).creator) }
```
Comment (tomlarkworthy) [inline samples/chat.bolt]: Oh this fixes a bug right?

File: samples/chat.bolt [hunk 2]
```diff
@@ -12,7 +12,10 @@ path /posts/{roomid}/{postid} is Timestamped<Post> {
 
 type RoomInfo {
   read() { isMember(this) || isUser(this.creator) }
-  write() { isInitial(this) || isUser(this.creator) }
+
```
Comment (tomlarkworthy) [inline samples/chat.bolt]: I think we can delete isInitial() from this file now.

**Comment by tomlarkworthy** [comment]
Looking good, is there a test for checking the error is thrown when you have both a write and create() on the same node?

File: samples/chat.bolt (hunk 1 above)
Comment (mckoss) [inline samples/chat.bolt]: Yes.  I had a faulty test too.

File: samples/chat.bolt (hunk 2 above)
Comment (mckoss) [inline samples/chat.bolt]: Still used in Created type.

**Comment by mckoss** [comment]
@tomlarkworthy Yes, error tests are [here](https://github.com/firebase/bolt/pull/121/files#diff-efac6a8073cbba997fd76d163cfe6b0eR425).

**Pull request merged**